### ISSUE TITLE
Update `GET /agents/summary/status` response to include agent group configuration sync status

### ISSUE THREAD

**Issue by vicferpoy** [wazuh/wazuh#12363]
## Description

As part of #9334 , we need to update this endpoint response to include a summary of the agent group configuration sync status.

Currently, this endpoint returns the following response:

```json
{
  "data": {
    "active": 1,
    "disconnected": 0,
    "never_connected": 2,
    "pending": 0,
    "total": 3
  },
  "error": 0
}
```

The proposed new response would be the following:

```json
{
  "data":{
    "connection":{
      "active": 2,
      "disconnected": 0,
      "never_connected": 5,
      "pending": 0,
      "total": 7 
    },
    "configuration":{
      "synced": 2
      "not_synced": 5
      "total": 7
    }
  },
  "error": 0
}
```

## Requirements

The only requirement for this development is that this needs to be implemented in the absolute most efficient way, as this endpoint will be used a lot in the WUI. A full refactor may be considered.

### wazuh/wazuh
- Unit tests without failures. Updated and/or expanded if there are new functions/methods/outputs:
  - [x] Cluster (`framework/wazuh/core/cluster/tests/` & `framework/wazuh/core/cluster/dapi/tests/`)
  - [x] Core (`framework/wazuh/core/tests/`)
  - [x] SDK (`framework/wazuh/tests/`)
  - [x] RBAC (`framework/wazuh/rbac/tests/`)
  - [x] API (`api/api/tests/`)
- API tavern integration tests without failures. Updated and/or expanded if needed (`api/test/integration/`):
  - [x] Affected tests 
  - [x] Affected RBAC (black and white) tests
- [x] Review integration test mapping using the script (`api/test/integration/mapping/integration_test_api_endpoints.json`)
- [x] Review of spec.yaml examples and schemas (`api/api/spec/spec.yaml`)
- [x] Review exceptions remediation when any endpoint path changes or is removed (`framework/wazuh/core/exception.py`)
- [x] Changelog (`CHANGELOG.md`)

### wazuh/wazuh-documentation
- [x] Migration from 3.X for changed endpoints (`source/user-manual/api/equivalence.rst`)
- [x] Update RBAC reference with new/modified actions/resources/relationships (`source/user-manual/api/rbac/reference.rst`)

**Comment by mcarmona99** [comment]
## Issue update

Unit and API integration tests have been updated according to the new changes following TDD. This issue will remain blocked until https://github.com/wazuh/wazuh/issues/12361 is solved.


Working branch: [`feature/12363-add-sync-agents-summary-status`](https://github.com/wazuh/wazuh/tree/feature/12363-add-sync-agents-summary-status)

**Comment by mcarmona99** [comment]
## Issue update


The framework function cannot improve more as it consists of a call to `WazuhDBQueryAgents` and a `for` loop counting the number of agents for each status. In spite of this, I have made tests in clusters with a big number of agents connected to test possible timeouts given by this endpoint. If the API gives any timeout, I will try to improve the endpoint functions. This test will be essential as this endpoint will be used a lot in the Wazuh UI.

### `GET /agents/summary/status` new times

**2 workers 10000 agents -> average 10 times = `0.393s`**

| #  | Time  |
|----|-----|
| 1               | 0.468s |
| 2                |  0.333s  | 
| 3                |  0.320s  | 
| 4                |  0.293s  | 
| 5                |  0.355s  | 
| 6                |  0.305s  | 
| 7                |  0.371s  | 
| 8                |  0.705s  | 
| 9                |  0.478s  | 
| 10               |  0.306s  | 
| **AVG**		| **0.393s** |


**2 workers 30000 agents -> average 10 times = `2.043s`**

| #  | Time  |
|----|-----|
| 1               | 2.416s |
| 2                |  1.716s  | 
| 3                |  2.566s  | 
| 4                |  2.654s  | 
| 5                |  2.164s  | 
| 6                |  1.530s  | 
| 7                |  1.505s  | 
| 8                |  2.304s  | 
| 9                |  1.529s  | 
| 10               |  2.041s  | 
| **AVG**		| **2.043s** |



**2 workers 40000 agents -> average 10 times = `2.575s`**

| #  | Time  |
|----|-----|
| 1               | 2.219s |
| 2                |  2.952s  | 
| 3                |  2.846s  | 
| 4                |  2.087s  | 
| 5                |  2.116s  | 
| 6                |  3.198s  | 
| 7                |  2.353s  | 
| 8                |  2.220s  | 
| 9                |  2.356s  | 
| 10               |  3.405s  | 
| **AVG**		| **2.575s** |




**5 workers 50000 agents -> average 10 times = `3.1572s`**


| #  | Time  |
|----|-----|
| 1               | 3.464s |
| 2                |  3.095s  | 
| 3                |   2.790s | 
| 4                | 3.164s   | 
| 5                |   3.175s | 
| 6                |  2.429s  | 
| 7                |  3.686s  | 
| 8                |  3.293s  | 
| 9                |  2.528s  | 
| 10               |  3.948s  | 
| **AVG**		| **3.1572S** |


In the following table, we can see the times spent in the endpoint depending on the number of agents connected to the cluster. Theoretically, the API timeout (10s) will be reached if there are 150000 agents connected (`500000÷3,1572=158368,1743`, `400000÷2,5752=155327,7415`, `300000÷2,0425=146878,825`, etc)

![image](https://user-images.githubusercontent.com/57481331/155334311-695bc390-eec8-445c-abd3-8ae388a7d5b0.png)



### `GET /agents/summary/status` profiling

I have measured times spent in the `get_agents_summary_status` framework function:

https://github.com/wazuh/wazuh/blob/cbcdcce87701537159d54bf7c1f580ae4e346c49/framework/wazuh/agent.py#L90-L121

**Iteration 1:**

Time spent between lines 106 and 111 (both included): `2.848771333694458s`
Time spent between lines 113 and 120 (both included): `0.05933332443237305s`

**Iteration 2:**

Time spent between lines 106 and 111 (both included): `2.4562878608703613s`
Time spent between lines 113 and 120 (both included): `0.034546613693237305s`

**Iteration 3:**

Time spent between lines 106 and 111 (both included): `2.5504984855651855s`
Time spent between lines 113 and 120 (both included): `0.031072139739990234s`

**Iteration 4:**

Time spent between lines 106 and 111 (both included): `1.913818597793579s`
Time spent between lines 113 and 120 (both included): `0.030908823013305664s`

**Iteration 5:**

Time spent between lines 106 and 111 (both included): `2.648632764816284s`
Time spent between lines 113 and 120 (both included): `0.029554367065429688s`


As we can see, the bottleneck is in the query. Therefore, the endpoint performance cannot be improved with changes in the framework function and we can say that `wazuh-db` is the daemon limiting this endpoint performance right now.
